pkgs/applications/radio/soapysdr/default.nix

```diff
@@ -35,6 +35,13 @@ in stdenv.mkDerivation {
     "-DCMAKE_BUILD_TYPE=Release"
   ] ++ lib.optional usePython "-DUSE_PYTHON_CONFIG=ON";
 
+  # https://github.com/pothosware/SoapySDR/issues/352
+  postPatch = ''
+    substituteInPlace lib/SoapySDR.in.pc \
+      --replace '$'{exec_prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
```

Comment:
@alexshpilkin

`''${something}` is the proper way to escape `${`

If that is what you were trying to do then I'll change them all to that

Reply:
No, that wasn’t it. This is more clever than I would like, but it has a purpose: there are two levels of escaping here, from Nix when it’s generating the script and from Bash when it’s executing it. That is to say, the file we’re patching contains the literal string `${exec_prefix}/@CMAKE_INSTALL_LIBDIR@` in it—the `${...}` is not shell syntax, it is pkg-config syntax. If you wanted to do this with standard Nix escaping as you suggest, you would end with `\''${exec_prefix}` or `'''${exec_prefix}'`, which is less subtle but even harder to parse I think.

(This is when .in files are being patched; take care that in some cases, CMake sources are being patched instead—so the interpolation is passing through Nix, Bash, and CMake on the way to the pkg-config file—and the CMake file being patched is already escaping the variable reference in one of a couple ways I’ve seen.)

If you see a clearer way, go for it, but the change you’re suggesting here wouldn’t work.

Reply:
Alright, do you think this is ready to merge?